packages/playback-core/src/request-errors.ts

```diff
@@ -0,0 +1,207 @@
+  const tokenNamePrefix = categoryToTokenNameOrPrefix(category);
+  const tokenName = `${tokenNamePrefix}Token` as const;
+  const token = isKeyOf(tokenName, muxMediaEl) ? muxMediaEl[tokenName] : undefined;
+  const expectedAud = categoryToAud(category);
+  const [playbackId] = toPlaybackIdParts(muxMediaEl.playbackId ?? '');
+  /** @TODO How to handle this case (CJP) */
+  // NOTE: *should* have playback id when reaching here
+  // if (!status) return MuxErrorCode.NO_STATUS;
+  if (!status || !playbackId) return undefined;
+
+  const jwtObj = parseJwt(token);
+  // Make sure we didn't get here because of a malformed JWT and/or claim
+  if (!!token && !jwtObj) {
+    console.error('malformed compact JWT DRM token string!');
+    /** @TODO Add error msg + context crud here (NOT YET DEFINED) (CJP) */
+    const mediaError = new MediaError();
+    mediaError.errorCategory = category;
+    mediaError.muxCode = MuxErrorCode.TOKEN_MALFORMED;
+    return mediaError;
+  }
+
+  if (status >= 500) {
+    /**
+     * @TODO We plausibly should have some basic retry logic for all other 500 status
+     * cases (CJP)
+     **/
+    console.error('generic server error!');
+    const mediaError = new MediaError('', mediaErrorCode, true);
+    mediaError.errorCategory = category;
+    mediaError.muxCode = MuxErrorCode.TOKEN_MALFORMED;
+    /** @TODO Add error msg + context crud here (NOT YET DEFINED) (CJP) */
+    return mediaError;
+  }
+
+  if (status === 403 || status === 400) {
+    if (jwtObj) {
+      if (isJWTExpired(jwtObj, requestTime)) {
+        const dateOptions: any = {
+          timeStyle: 'medium',
+          dateStyle: 'medium',
+        };
+        const message = i18n(`The video’s secured ${tokenNamePrefix}-token has expired.`, translate);
+        /** @TODO move lang build crud from mux-player to playback-core (See: esbuilder.js) (CJP) */
+        const context = i18n(`Expired at: {expiredDate}. Current time: {currentDate}.`, translate).format({
+          // expiredDate: new Intl.DateTimeFormat(lang.code, dateOptions).format(tokenExpiry * 1000),
+          // currentDate: new Intl.DateTimeFormat(lang.code, dateOptions).format(Date.now()),
+          expiredDate: new Intl.DateTimeFormat('en', dateOptions).format(jwtObj.exp ?? 0 * 1000),
+          currentDate: new Intl.DateTimeFormat('en', dateOptions).format(requestTime),
+        });
+        const mediaError = new MediaError(message, mediaErrorCode, true, context);
+        mediaError.errorCategory = category;
+        mediaError.muxCode = MuxErrorCode.TOKEN_MALFORMED;
+        return mediaError;
+      }
+      if (isJWTSubMismatch(jwtObj, playbackId)) {
+        const message = i18n(
+          `The video’s playback ID does not match the one encoded in the ${tokenNamePrefix}-token.`,
+          translate
+        );
+        const context = i18n(
+          `Specified playback ID: {playbackId} and the playback ID encoded in the ${tokenNamePrefix}-token: {tokenPlaybackId}`,
```

Comment:
NOTE: Need to rework this and other strings to parameterize based on the request type/"category" (e.g. `drm-token` vs. `playback-token`).